content/blog/2020-06-29-june-20-community-gems copy.md

```diff
@@ -0,0 +1,144 @@
+```
+
+Read up on configuring supported storage
+[in our docs](https://dvc.org/doc/command-reference/remote/add#supported-storage-types).
+
+### Q: [If I have a folder containing many data files, is there any advantage to zipping the folder and DVC tracking the `.zip`?](https://discord.com/channels/485586884165107732/563406153334128681/714922184455225445)
+
+There are a few things to consider:
+
+- **CPU time.** Even though it can be faster to pull a single file than a
+  directory (though not in all cases, since we can parallelize directory
+  downloads), the tradeoff is the time needed to unzip your data. Depending on
+  your constraints, this can be expensive and undesirable.
+
+- **Deduplication.** DVC deduplicates on the file level. So if you add one
+  single file to a directory, DVC will save only that file, not the whole
+  dataset again. If you use a zipped directory you won't get this benefit. In
+  the long run, this could be more expensive in terms of storage space for your
+  DVC cache and remote if the contents of your dataset change frequently.
+
+Generally, we would recommend first trying a plain unzipped directory. DVC is
+designed to work with large numbers of files (on the order of millions) and has
+the latest release (DVC 1.0) has
+[optimizations built for this purpose exactly](https://dvc.org/blog/dvc-1-0-release#data-transfer-optimizations).
+
+### [Q: Can I execute a `dvc push` with the DVC Python API inside a Python script?](https://discord.com/channels/485586884165107732/485596304961962003/718419219288686664)
+
+Currently, our [Python API](https://dvc.org/doc/api-reference#python-api)
+doesn't support commands like `dvc push`,`dvc pull`, or `dvc status`. It is
+designed for interfacing with objects tracked by DVC. That said, CLI commands
+are basically calling `dvc.repo.Repo` object methods. So if you want to use
+commands from within Python code, you could try creating a `Repo` object with
+`r = Repo({root_dir})` and then `r.push()`. Please note that we don't officially
+support this use case yet.
+
+Of course, you can also run DVC commands from a Python script using `sys` or a
+similar library for issuing system commands.
+
+### [Q: Does the `dvc pipeline` command for visualizing pipelines still work in DVC 1.0?](https://discord.com/channels/485586884165107732/485596304961962003/717682556203565127)
+
+Most of the `dvc pipeline` functionality- like `dvc pipeline show --ascii` to
+print out an ASCII diagram of your pipeline- has been migrated to a new command,
+`dvc dag`. This function is written for our new pipeline format. Check out
+[our new docs](https://dvc.org/doc/command-reference/dag#dag) for an example.
+
+### [Q: Is there a way to create a DVC pipeline stage without running the commands in that stage?](https://discord.com/channels/485586884165107732/485596304961962003/715271980978405447)
+
+Yes. Say you have a Python script, `train.py`, that takes in a dataset `data`
```

Comment:
Also, we can reference again the `dvc.yaml` docs and clarify that the file can be edited manually - e.g. copy/paste existing stage and modify it.

Reply:
done